packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_set_loan_token.test.ts

```diff
@@ -0,0 +1,369 @@
+import { WIF } from '@defichain/jellyfish-crypto'
+import BigNumber from 'bignumber.js'
+import { LoanMasterNodeRegTestContainer } from './loan_container'
+import { Testing } from '@defichain/jellyfish-testing'
+import { RegTest } from '@defichain/jellyfish-network'
+
+const container = new LoanMasterNodeRegTestContainer()
+const testing = Testing.create(container)
+
+let providers: MockProviders
+let builder: P2WPKHTransactionBuilder
+
+describe('loan.setLoanToken()', () => {
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+
+    providers = await getProviders(testing.container)
+    providers.setEllipticPair(WIF.asEllipticPair(GenesisKeys[GenesisKeys.length - 1].owner.privKey))
+    builder = new P2WPKHTransactionBuilder(providers.fee, providers.prevout, providers.elliptic, RegTest)
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  beforeEach(async () => {
+    // Fund 10 DFI UTXO
+    await fundEllipticPair(testing.container, providers.ellipticPair, 10)
+    await providers.setupMocks() // Required to move utxos
+  })
+
+  it('should setLoanToken', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token1',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setLoanToken({
+      symbol: 'Token1',
+      name: 'Token1',
+      priceFeedId,
+      mintable: false,
+      interest: new BigNumber(1.5)
+    }, script)
+
+    // Ensure the created txn is correct.
+    const outs = await sendTransaction(testing.container, txn)
+    expect(outs[0].value).toStrictEqual(0)
+    expect(outs[1].value).toBeLessThan(10)
+    expect(outs[1].value).toBeGreaterThan(9.999)
+    expect(outs[1].scriptPubKey.addresses[0]).toStrictEqual(await providers.getAddress())
+
+    const loanTokenId = calculateTxid(txn)
+    const data = await testing.container.call('listloantokens', [])
+    expect(data).toStrictEqual({
+      [loanTokenId]: {
+        token: {
+          1: {
+            symbol: 'Token1',
+            symbolKey: 'Token1',
+            name: 'Token1',
+            decimal: 8,
+            limit: 0,
+            mintable: false,
+            tradeable: true,
+            isDAT: true,
+            isLPS: false,
+            finalized: false,
+            isLoanToken: true,
+            minted: 0,
+            creationTx: loanTokenId,
+            creationHeight: await testing.container.getBlockCount(),
+            destructionTx: '0000000000000000000000000000000000000000000000000000000000000000',
+            destructionHeight: -1,
+            collateralAddress: expect.any(String)
+          }
+        },
+        priceFeedId,
+        interest: 1.5
+      }
+    })
+  })
+
+  // NOTE(jingyi2811): There are bugs in the C++ side
```

Comment:
Mihailo is fixing this. Currently, you can still pass in symbol which is an empty string or more than 8 letters and C++ side will return you other error which is not related